apollo-router/src/plugins/traffic_shaping/retry.rs

```diff
@@ -67,12 +71,12 @@ impl<Res, E> Policy<subgraph::Request, Res, E> for RetryPolicy {
                     subgraph = %self.subgraph_name,
                 );
 
-                Some(future::ready(self.clone()))
+                Some(future::ready(()))
```

Comment:
instead of returning a new policy, you can now mutate the policy inside this function, so there is no need to return `self.clone()`.